preferences/src/androidTest/java/com/eis/preferences/PreferencesManagerTest.java

```diff
@@ -0,0 +1,162 @@
+
+import android.content.Context;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+
+/**
+ * @author Luca Crema
+ */
+public class PreferencesManagerTest {
+
+    private static final String DEFAULT_INT_KEY = "defaultIntKey";
+    private static final String DEFAULT_STRING_KEY = "defaultStringKey";
+    private static final String DEFAULT_BOOL_KEY = "defaultBooleanKey";
+    private static final String DEFAULT_OBJECT_KEY = "defaultObjectKey";
+    private static final int DEFAULT_INT_VALUE = 100;
+    private static final String DEFAULT_STRING_VALUE = "Roberto";
+    private static final boolean DEFAULT_BOOL_VALUE = true;
+    private static final String DEFAULT_OBJECT_VALUE = "Test String";
+    private static final int MAX_SHIFT_VALUE = 3;
+    private Context ctx;
+
+    @Before
+    public void setUp() {
+        ctx = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        PreferencesManager.removeAllValues(ctx);
+    }
+
+    @Test
+    public void getInt_defaultInteger_isEquals() {
+        Assert.assertEquals(PreferencesManager.DEFAULT_INTEGER_RETURN, PreferencesManager.getInt(ctx, DEFAULT_INT_KEY));
+    }
+
+    @Test
+    public void getString_defaultString_isEquals() {
+        Assert.assertEquals(PreferencesManager.DEFAULT_STRING_RETURN, PreferencesManager.getString(ctx, DEFAULT_STRING_KEY));
+    }
+
+    @Test
+    public void getObject_isNull() {
+        Assert.assertNull(PreferencesManager.getObject(ctx, DEFAULT_OBJECT_KEY));
+    }
+
+    @Test
+    public void getBoolean_defaultBoolean_isEquals() {
+        Assert.assertEquals(PreferencesManager.DEFAULT_BOOLEAN_RETURN, PreferencesManager.getBoolean(ctx, DEFAULT_BOOL_KEY));
+    }
+
+    @Test
+    public void setInt_getInt_isEquals() {
+        PreferencesManager.setInt(ctx, DEFAULT_INT_KEY, DEFAULT_INT_VALUE);
+        Assert.assertEquals(DEFAULT_INT_VALUE, PreferencesManager.getInt(ctx, DEFAULT_INT_KEY));
+    }
+
+    @Test
+    public void setString_getString_isEquals() {
+        PreferencesManager.setString(ctx, DEFAULT_STRING_KEY, DEFAULT_STRING_VALUE);
+        Assert.assertEquals(DEFAULT_STRING_VALUE, PreferencesManager.getString(ctx, DEFAULT_STRING_KEY));
+    }
+
+    @Test
+    public void setBoolean_getBoolean_isEquals() {
+        PreferencesManager.setBoolean(ctx, DEFAULT_BOOL_KEY, DEFAULT_BOOL_VALUE);
+        Assert.assertEquals(DEFAULT_BOOL_VALUE, PreferencesManager.getBoolean(ctx, DEFAULT_BOOL_KEY));
+    }
+
+    @Test
+    public void setObject_getBObject_isEquals() {
+        try {
+            PreferencesManager.setObject(ctx, DEFAULT_OBJECT_KEY, DEFAULT_OBJECT_VALUE);
+        } catch (IOException e) {
+            Assert.fail("Should not have thrown an IOException");
+        }
+        Assert.assertEquals(DEFAULT_OBJECT_VALUE, PreferencesManager.<String>getObject(ctx, DEFAULT_OBJECT_KEY));
```

Comment:
I ran this test on my machine and it failed at line 80:
```
java.lang.AssertionError: expected:<Test String> but was:<null>
at org.junit.Assert.fail(Assert.java:88)
at org.junit.Assert.failNotEquals(Assert.java:834)
at org.junit.Assert.assertEquals(Assert.java:118)
at org.junit.Assert.assertEquals(Assert.java:144)
at com.eis.preferences.PreferencesManagerTest.setObject_getBObject_isEquals(PreferencesManagerTest.java:80)
```

Reply:
Forgot to clean the PreferencesManager class, the getObject and setObject were not ready to release yet. I removed the two methods and the testing methods